packages/abi-typegen/src/types/interfaces/JsonAbiNew.ts

```diff
@@ -0,0 +1,101 @@
+/**
+ * Types for Fuel JSON ABI Format as defined on:
+ * https://github.com/FuelLabs/fuel-specs/blob/master/src/abi/json-abi-format.md
+ */
+export interface JsonAbi {
+  readonly specVersion: string;
+  readonly encodingVersion: string;
+  readonly programType: string;
+  readonly concreteTypes: readonly ConcreteType[];
+  readonly metadataTypes: readonly MetadataType[];
+  readonly functions: readonly AbiFunction[];
+  readonly loggedTypes: readonly LoggedType[];
+  readonly messagesTypes: readonly MessageType[];
+  readonly configurables: readonly Configurable[];
+}
+
+export interface ConcreteType {
+  readonly type: string;
+  readonly concreteTypeId: string;
+  readonly metadataTypeId?: number;
+  readonly typeArguments?: readonly string[];
+}
+
+export interface MetadataType {
+  readonly type: string;
+  readonly metadataTypeId: number;
+  readonly components?: readonly Component[];
+  readonly typeParameters?: readonly number[];
+}
+
+export interface Component extends TypeArgument {
+  readonly name: string;
+}
+
+export interface TypeArgument {
+  readonly typeId: number | string; // the type metadata declaration ID or type concrete declaration hash based ID of the type of the component.
+  readonly typeArguments?: readonly TypeArgument[];
+}
+
+export interface AbiFunction {
+  readonly name: string;
+  readonly inputs: readonly AbiFunctionInput[];
+  readonly output: string;
+  readonly attributes: readonly AbiFunctionAttribute[] | null;
```

Comment:
```suggestion
  readonly attributes?: readonly AbiFunctionAttribute[];
```

The specification mentions "an optional array of attributes". Should this be optional and not nullable?